test/dynamo/test_repros.py

```diff
@@ -3582,6 +3582,7 @@ def f(x):
         with torch._dynamo.config.patch(error_on_nested_jit_trace=False):
             torch.jit.trace(f, torch.randn(3))
 
+    @unittest.skip("TODO: broken by https://github.com/pytorch/pytorch/pull/126905")
```

Comment:
This is the offline solver cluster of issues

Reply:
The error on this one is

```
======================================================================
FAIL: test_tensor_split (__main__.ReproTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/data/users/ezyang/a/pytorch/torch/testing/_internal/common_utils.py", line 2741, in wrapper
    method(*args, **kwargs)
  File "/home/ezyang/local/a/pytorch-env/lib/python3.10/contextlib.py", line 79, in inner
    return func(*args, **kwds)
  File "/data/users/ezyang/a/pytorch/test/dynamo/test_repros.py", line 3594, in test_tensor_split
    gm, _ = torch._dynamo.export(
  File "/data/users/ezyang/a/pytorch/torch/_dynamo/eval_frame.py", line 1332, in inner
    dim_constraints.solve()
  File "/data/users/ezyang/a/pytorch/torch/fx/experimental/symbolic_shapes.py", line 1773, in solve
    assert isinstance(solution, sympy.Eq), f"Expected an equality constraint for {s}, got {solution}"
AssertionError: Expected an equality constraint for s0, got False
```

Reply:
The equations being solved: `{((s0//2) <= s0) & (-s0 <= (s0//2)), 0 <= s0 - (s0//2), s0 - (s0//2) >= 0, Eq(((s0 + (s0//2) - 3)//((s0//2))), 2), s0 >= 2, s0 >= (s0//2), (s0//2) <= s0}`

Reply:
On main, the equations we get are `reduce_inequalitites {Eq(1 + 2*(s0 - 3)/s0, 2), s0 >= 2} s0`

Reply:
So it looks like this is a consequence of removing the x // y --> floor(x / y) rule? It's kind of weird that the floor was able to be removed in the original though...

Reply:
It looks like the general issue is sympy doesn't know how to handle CleanDiv instead of the div op. The first issue is some of those equations should have statically resolved to True/False (e.g. `s0 - (s0//2) >= 0`), but stay in that form with CleanDiv, so too many expressions get passed to the export solver.

But even if we fix remove those trivial exprs, the expressions that shouldn't resolve (e.g. `Eq(1 + 2*(s0 - 3)/s0, 2)`) crash because the export's solver call to `sympy.solvers.inequalities.reduce_inequalities(exprs, s)` doesn't know what to do with CleanDiv, so we get an error like:
```
NotImplementedError: 
The inequality, Eq(((x + (x//2) - 3)//((x//2))) - 2, 0), cannot be
solved using solve_univariate_inequality.
```

Could we just fallback to normal div for export's solver? Or is there some way of adding support so sympy does something with CleanDiv

Reply:
So something like this, but I am pretty sure I had a good reason for replacing this with CleanDiv in the first place

```
diff --git a/test/dynamo/test_repros.py b/test/dynamo/test_repros.py
index 88a6492beea..1812b7fe0e4 100644
--- a/test/dynamo/test_repros.py
+++ b/test/dynamo/test_repros.py
@@ -3586,7 +3586,6 @@ class ReproTests(torch._dynamo.test_case.TestCase):
         with torch._dynamo.config.patch(error_on_nested_jit_trace=False):
             torch.jit.trace(f, torch.randn(3))
 
-    @unittest.skip("TODO: broken by https://github.com/pytorch/pytorch/pull/126905")
     @torch._dynamo.config.patch("assume_static_by_default", False)
     def test_tensor_split(self):
         def f(x):
diff --git a/torch/fx/experimental/symbolic_shapes.py b/torch/fx/experimental/symbolic_shapes.py
index 7247aabfaf5..f52484e7fb5 100644
--- a/torch/fx/experimental/symbolic_shapes.py
+++ b/torch/fx/experimental/symbolic_shapes.py
@@ -1612,7 +1612,10 @@ class DimConstraints:
             congruence = (base - mod_reduced) % divisor
             if congruence != 0:
                 self._congruences[s].add(congruence)
-            return CleanDiv(base - mod_reduced, divisor)
+            # NB: Must not be CleanDiv, it needs to be regular sympy division
+            # so inequality solver works.  This is sort of problematic for
+            # is_integer tests though haha
+            return (base - mod_reduced) / divisor
 
         if expr.has(Mod):
             expr = expr.replace(Mod, mod_handler)
```